pkg/detectors/ftp/ftp_test.go

```diff
@@ -95,9 +135,14 @@ func TestFTP_FromChunk(t *testing.T) {
 			// }
 			for i := range got {
 				got[i].Raw = nil
+
+				if (got[i].VerificationError != nil) != tt.wantVerificationErr {
+					t.Fatalf("wantVerificationError = %v, verification error = %v", tt.wantVerificationErr, got[i].VerificationError)
+				}
 			}
-			if diff := pretty.Compare(got, tt.want); diff != "" {
-				t.Errorf("URI.FromData() %s diff: (-got +want)\n%s", tt.name, diff)
+			opts := cmpopts.IgnoreFields(detectors.Result{}, "VerificationError")
+			if diff := cmp.Diff(got, tt.want, opts); diff != "" {
+				t.Errorf("FTP.FromData() %s diff: (-got +want)\n%s", tt.name, diff)
 			}
```

Comment:
**Nit / question:** Could we add `"Raw"` to the `cmpopts.IgnoreFields` and remove setting it to `nil` on line 137? If so, we could convert that loop into iterating over values also.

Reply:
oh yeah probably

Reply:
good thought moving `Raw` into the ignorelist, but i don't follow the second part of your suggestion

Reply:
I think the only reason we were iterating over the indices was to clear out the `Raw` value in the object, so now that `Raw` is being ignored we don't need to do that and could update the loop to be:

```go
for _, gotResult := range got {
    if (gotResult.Verification != nil) != tt.wantVerificationErr {
        // ...
    }
}
```

Reply:
oh, no, it's dumber than that. the root problem is that we know when we want to see a verification error, but we don't always know the exact shape of the error, so there's a top-level "is there an error" flag. but verification errors are per [i]result[/i], not per overall chunk flow, so we loop through and check the expected flag against each individual result object.

there's probably a better way to do it, though - i haven't thought about it too hard.